docs/howto/trace-the-charm-code.md

```diff
@@ -5,36 +5,75 @@
-- How and when to use the lower-level API
+Refer to the `ops.tracing` reference for the canonical usage example, configuration
+options, and API details.
 
 ## Getting started
 
 To enable basic tracing:
 
-- In your charm's `pyproject.toml` or `requirements.txt`, add `ops[tracing]` dependency
-- In your `charmcraft.yaml`, declare the tracing and (optionally) ca relations
-- In your charm class `__init__`, instantiate the `ops.tracing.Tracing(...)` object
-
-At this point, the Ops library will be traced:
+- In your charm's `pyproject.toml` or `requirements.txt`, add `ops[tracing]` as a dependency
+- In your `charmcraft.yaml`, declare the tracing and (optionally) ca relations, for example:
+
+```yaml
+requires:
+  charm-tracing:
+    interface: tracing
+    limit: 1
+    optional: true
+  receive-ca-cert:
+    interface: certificate_transfer
+    limit: 1
+    optional: true
+```
+
+- In your charm class `__init__`, instantiate the `ops.tracing.Tracing(...)` object, for example:
+
+```python
+class MyCharm(ops.CharmBase):
+    def __init__(self, framework: ops.Framework):
+        super().__init__(framework)
+        self.tracing = ops.tracing.Tracing(
+            self,
+            tracing_relation_name='charm-tracing',
+            ca_relation_name='receive-ca-cert',
+        )
+        ...
+```
+
+At this point, Ops will trace:
 - The `ops.main()` call
-- The events that the Ops library emits
-- The hook tools called by the charm code
-- The Pebble API access by the charm code
+- Events that Ops emits, including all the Juju events
+- Hook tools called by the charm code
+- Pebble API access by the charm code
 
 This provides coarse-grained tracing, focused on the boundaries between the
 charm code and the external processes.
 
+When you deploy your charm, until it is integrated with an app providing the `tracing` relation
+(and optionally the `certificate_transfer` relation), the traces will be buffered in a tracing
+database on the unit. When the charm is successfully integrated with the `tracing` provider,
+the buffered traces and new traces will be sent to the tracing destination. For Kubernetes
+charms, if the container is recreated, any buffered traces will be lost. Ops will buffer traces
+for a reasonable period of time and amount of space.
+
+For example, to send traces to [Grafana Tempo](https://grafana.com/docs/tempo/latest/) from
+a charm named `my-charm`, assuming that
+[Charmed Tempo HA](https://discourse.charmhub.io/t/charmed-tempo-ha/15531) has already been
+deployed:
+
+TODO: what are the actual names for tempo's tracing and certificate_transfer?
+
+```bash
+juju deploy my-charm
+juju integrate my-charm tempo
+```
+
 (custom-spans-and-events)=
 ## Custom spans and events
 
+TODO: why do we need to depend on opentelemetry when ops does? Is it the newer version? If so why not just have ops depend on that?
+
 - In your charm's `pyproject.toml` or `requirements.txt`, add
   `opentelemetry-api ~= 1.30.0` dependency.
```

Comment:
@dimaqq do you actually have to add `opentelemetry-api ~= 1.30.0` as a dependency? Or will the opentelemetry-api dependency in ops provide this for you?

Reply:
`ops[tracing]` will bring that in, sure.

I was thinking "explicit is better than implicit" though.
Perhaps this can be footnoted instead, like "if you need to control the opentelementry API version, do this..."